File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
import java.util.stream.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String, Long> pMap = Arrays.stream(participant)
                         .collect(Collectors.groupingBy(i->i,Collectors.counting()) );
        //System.out.println("pMap : " +pMap);
        
        for (String s : completion) {
            Long put = pMap.put(s, pMap.get(s) - 1);
        }
        
        //System.out.println("pMap : " +pMap);

        return pMap.entrySet()
            .stream().filter(entry->entry.getValue().equals(1L)).map(entry->entry.getKey()).findAny().get();
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import *
def solution(participant, completion):
    answer = ''
    
    p_counter = Counter(participant)
    
    for p in completion:
        
        if p in p_counter:
            p_counter[p] -= 1
            if p_counter[p] == 0:
                del p_counter[p]
    
    print(p_counter)
    
    return [ key for key in p_counter][0]